includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1055,6 +1055,49 @@ public function generate_create_intent_request( $order, $prepared_source ) {
 		return apply_filters( 'wc_stripe_generate_create_intent_request', $request, $order, $prepared_source );
 	}
 
+	/**
+	 * Create the level 3 data array to send to Stripe when making a purchase.
+	 *
+	 * @param WC_Order $order           The order that is being paid for.
+	 * @param string   $store_postcode  The postcode of the store.
+	 * @return array                    The level 3 data to send to Stripe.
+	 */
+	public function get_level3_data_from_order( $order, $store_postcode ) {
+		// Get the order items. Don't need their keys, only their values.
+		// Order item IDs are used as keys in the original order items array.
+		$order_items = array_values( $order->get_items() );
+		$currency    = $order->get_currency();
+
+		$stripe_line_items = array_map(function( $item ) use ( $currency ) {
+			$product_id          = $item->get_variation_id()
+				? $item->get_variation_id()
+				: $item->get_product_id();
+			$product             = wc_get_product( $product_id );
```

Comment:
`WC_Order_Item_Product::get_product()` does something very similar (check for a variation, load the product, etc.). Any reason not to use the method?

Reply:
To follow up here, it's using the item directly now 👍